src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -77,6 +87,20 @@ const MetricsSelector = ({ maxSelectionNumber, metrics, name, onSelect, preselec
     isOpen && 'metrics-selector-header_open'
   )
 
+  useEffect(() => {
+    if (!modelEndpointUid) return
+    selectedModelEndpointUid.current = modelEndpointUid
+    appliedMetricsRef.current = preselectedMetrics
+  }, [modelEndpointUid, preselectedMetrics])
+
+  useEffect(() => {
+    if (isOpen) {
+      formRef.current.reset({
+        metrics: appliedMetricsRef.current.map(metricItem => metricItem.full_name)
+      })
+    }
```

Comment:
That's not correct to use `reset` method. Because it changes the initial state of the form.
Also it's better to revert selected metrics on closing the pop up and not the opening. And set the new list of metrics on pressing the Apply button

Reply:
"reset" changed to "change"
selected metrics applied on closing popup. 
change form state on apply.